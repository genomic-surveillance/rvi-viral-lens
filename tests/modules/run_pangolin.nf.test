nextflow_process {

    name "Test Process run_pangolin"
    script "modules/run_pangolin.nf"
    process "run_pangolin"

    test("Should run without failures") {

        when {
            params {
                // define parameters here. Example:
                meta = [
                    id:"test.2222222",
                    sample_id:"test",
                    taxid:2222222
                ]
                test_loc="${projectDir}/tests/test_data"
            }
            process {
                """
                // define inputs of the process here. Example:
                input[0] = tuple(
                    params.meta,
                    "${params.test_loc}/test_consensuses/test2_sarscov2.fa"
                )
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
