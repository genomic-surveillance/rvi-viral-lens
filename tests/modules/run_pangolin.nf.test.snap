{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test.2222222",
                            "taxid": 2222222,
                            "sample_id": "test"
                        },
                        "test2_sarscov2.fa:md5,1f41369aa84e4e348f0b5ba6d70a7257",
                        "EG.5.1.1"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "22.04.5"
        },
        "timestamp": "2024-02-29T12:02:44.185278"
    }
}